src/rgw/rgw_amqp_1.cc

```diff
@@ -0,0 +1,486 @@
+  static const int AMQP_1_TRACKER_STATUS_RECEIVED = -0x3002;
+  static const int AMQP_1_TRACKER_STATUS_ACCEPTED = -0x3003;
+  static const int AMQP_1_TRACKER_STATUS_REJECTED = -0x3004;
+  static const int AMQP_1_TRACKER_STATUS_RELEASED = -0x3005;
+  static const int AMQP_1_TRACKER_STATUS_MODIFIED = -0x3006;
+
+  static const int RGW_AMQP_1_STATUS_OK = 0x0;
+
+  int status_tracker_to_rgw(int s) {
+    if(s == AMQP_1_TRACKER_STATUS_ACCEPTED) {
+      return RGW_AMQP_1_STATUS_OK;
+    } else {
+      return s;
+    }
+  }
+
+  struct reply_callback_with_tracker_t {
+    proton::tracker tracker;
+    reply_callback_t cb;
+
+    reply_callback_with_tracker_t(proton::tracker _tracker, reply_callback_t
+        _cb) : tracker(_tracker), cb(_cb) { }
+
+    bool operator==(proton::tracker rhs) const {
+      return tracker == rhs;
+    }
+  };
+
+  struct message_wrapper_t {
+    connection_ptr_t conn;
+    std::string topic;
+    std::string message;
+    reply_callback_t cb;
+
+    message_wrapper_t(connection_ptr_t& _conn,
+      const std::string& _topic,
+      const std::string& _message,
+      reply_callback_t _cb) : conn(_conn), topic(_topic), message(_message), cb(_cb) { }
+  };
+
+
+  typedef std::unordered_map<std::string, connection_ptr_t> ConnectionList;
+  typedef boost::lockfree::queue<message_wrapper_t*, boost::lockfree::fixed_sized<true>> MessageQueue;
+  typedef std::vector<reply_callback_with_tracker_t> CallbackList;
+
+  // proton::tracker states to string
+  std::string to_string(enum proton::transfer::state s) {
+    switch(s) {
+      case proton::transfer::state::NONE:
+        return "AMQP_1_TRACKER_STATUS_UNKOWN";
+      case proton::transfer::state::RECEIVED:
+        return "AMQP_1_TRACKER_STATUS_RECEIVED";
+      case proton::transfer::state::ACCEPTED:
+        return "AMQP_1_TRACKER_STATUS_ACCEPTED";
+      case proton::transfer::state::REJECTED:
+        return "AMQP_1_TRACKER_STATUS_REJECTED";
+      case proton::transfer::state::RELEASED:
+        return "AMQP_1_TRACKER_STATUS_RELEASED";
+      case proton::transfer::state::MODIFIED:
+        return "AMQP_1_TRACKER_STATUS_MODIFIED";
+    }
+    return "";
+  }
+
+  int reply_to_code(enum proton::transfer::state s) {
+    switch(s) {
+      case proton::transfer::state::NONE:
+        return AMQP_1_TRACKER_STATUS_UNKOWN;
+      case proton::transfer::state::RECEIVED:
+        return AMQP_1_TRACKER_STATUS_RECEIVED;
+      case proton::transfer::state::ACCEPTED:
+        return AMQP_1_TRACKER_STATUS_ACCEPTED;
+      case proton::transfer::state::REJECTED:
+        return AMQP_1_TRACKER_STATUS_REJECTED;
+      case proton::transfer::state::RELEASED:
+        return AMQP_1_TRACKER_STATUS_RELEASED;
+      case proton::transfer::state::MODIFIED:
+        return AMQP_1_TRACKER_STATUS_MODIFIED;
+    }
+    return AMQP_1_TRACKER_STATUS_UNKOWN;
+  }
+
+  static const size_t MAX_QUEUE_DEFAULT = 1024;
+
+  class connection_t : public proton::messaging_handler{
+    public:
```

Comment:
are these parameters accessed from the outside?
if not, please make them "private"